api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -191,6 +203,63 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
     return ResponseEntity.ok(createdWorkspace);
   }
 
+  @Override
+  public ResponseEntity<WorkspaceOperation> createWorkspaceAsync(Workspace workspace) {
+    // Basic request validation.
+    validateWorkspaceApiModel(workspace);
+    getLiveCdrVersionId(workspace.getCdrVersionId());
+
+    // TODO: enforce access level check here? Not strictly necessary, but may make sense as
+    // belt/suspenders check.
+
+    Timestamp now = new Timestamp(clock.instant().toEpochMilli());
+    DbWorkspaceOperation operation =
+        workspaceOperationDao.save(
+            new DbWorkspaceOperation()
+                .setCreatorId(userProvider.get().getUserId())
+                .setStatus(DbWorkspaceOperationStatus.PENDING)
+                .setCreationTime(now)
+                .setLastModifiedTime(now));
+
+    taskQueueService.pushCreateWorkspaceTask(operation.getId(), workspace);
+    return ResponseEntity.ok(workspaceOperationMapper.toModelWithoutWorkspace(operation));
+  }
+
+  @Override
+  public ResponseEntity<WorkspaceOperation> getWorkspaceOperation(Long id) {
+    return workspaceOperationDao
+        .findById(id)
+        .filter(op -> op.getCreatorId() == userProvider.get().getUserId())
+        .map(op -> workspaceOperationMapper.toModelWithWorkspace(op, workspaceDao, workspaceMapper))
```

Comment:
Hm.. I'd thought this mapping might require a Terra GetWorkspace call as well. Is that somehow happening indirectly in here?

Reply:
It is not, but surprisingly that only means we're missing `googleBucketName`

but you're right, I should fix this

Reply:
done